beater/api/config/agent/handler.go

```diff
@@ -97,6 +97,7 @@ func Handler(client kibana.Client, config *config.AgentConfig) request.Handler {
 		// configuration successfully fetched
 		c.Header().Set(headers.CacheControl, cacheControl)
 		c.Header().Set(headers.Etag, fmt.Sprintf("\"%s\"", result.Source.Etag))
+		c.Header().Set("Access-Control-Expose-Headers", headers.Etag)
```

Comment:
nit: could you add the `"Access-Control-Expose-Headers"` to the [headers/keys](https://github.com/elastic/apm-server/blob/master/beater/headers/keys.go), for reuse (it is also used in the CORS middleware)